signup/forms.py

```diff
@@ -118,15 +122,22 @@ class PasswordUpdateForm(PasswordConfirmMixin, forms.ModelForm):
 
     class Meta:
         model = User
-        fields = ['new_password', 'new_password2']
+        fields = ['password', 'new_password', 'new_password2']
 
     def __init__(self, *args, **kwargs):
         self.user = kwargs.pop('instance')
         super(PasswordUpdateForm, self).__init__(*args, **kwargs)
 
+    def clean_password(self):
+        password = self.cleaned_data['password']
+        pwd_correct = self.user.check_password(password)
```

Comment:
Why do we do the `check_password` here if it is also done later on in the processing of the form/serializer data?

Reply:
Where exactly?